src/Controllers/ElementController.php

```diff
@@ -68,10 +68,14 @@ public function getElement()
      * Renders the managed {@link BaseElement} wrapped with the current
      * {@link ElementController}.
      *
-     * @return string HTML
+     * @return null|string HTML
      */
     public function forTemplate()
     {
+        if (!$this->element->canView()) {
+            return null;
+        }
```

Comment:
I think I'd rather put `<% if $canView %>` checks into the template, e.g. around ElementHolder.ss

If continuing with this approach we'd need to update the unit test that's failing

Reply:
We’ve had issues in core before, I think with `Versioned::doPublish()`, where people bypass permission checks intentionally (e.g. CLI publishes and stuff like that) and expect things to work. I think this could fall into that category - if I call `$html = $element->forTemplate()` directly, I wouldn’t expect `null`.

I’d suggest filtering the elements in `ElementalArea::ElementControllers()`, something like

```php
$items = $this->Elements()->filterByCallback(function (BaseElement $item) {
    return $item->canView();
});
```